### PR TITLE
&felipeangelimvieira, yarnabrina [MNT] CI step to test that notebooks run

### PR DESCRIPTION
**Pull request by fkiraly**
This adds a CI step to run all the tutorial notebooks.

The test passes if all tutorial notebooks run without error, and can be used to test the fixes in https://github.com/jdb78/pytorch-forecasting/pull/1615

### CODE REVIEW

**Comment by fkiraly** [comment]
FYI @XinyuWuu, @yarnabrina, any ideas why the GHA cannot see the sh script?

**Comment by fkiraly** [comment]
credits:

```

[tool.setuptools]
py-modules = ["pytorch_forecasting"]

[build-system]
build-backend = "setuptools.build_meta"
requires = [
  "setuptools>61",
]
```

suggestio due to @felipeangelim

**Comment by fkiraly** [comment]
oh, still permission denied

**Comment by yarnabrina** [comment]
I think you haven't changed file level permission, e.g. with chmod. Do you want me to push a test commit in this branch?

**Comment by fkiraly** [comment]
> I think you haven't changed file level permission, e.g. with chmod. Do you want me to push a test commit in this branch?

would be appreciated!

File: pyproject.toml [hunk 1]
```diff
@@ -100,3 +100,12 @@ dev = [
...
 graph = ["networkx"]
 mqf2 = ["cpflows"]
+
+[tool.setuptools]
+py-modules = ["pytorch_forecasting"]
+
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = [
+  "setuptools>61",
```
Comment (yarnabrina) [inline pyproject.toml]: I think a new vulnerability was recently discovered in `setuptools` (ref. CVE-2024-6345), a newer version may be better.

```suggestion
  "setuptools>=70.0.0",
```

**Comment by fkiraly** [comment]
Nice! The test now runs as intended!

The failure is genuine and has identical root cause with https://github.com/jdb78/pytorch-forecasting/issues/1614

**Pull request merged**